src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -152,188 +164,159 @@ public bool ParseHeaders(TRequestHandler handler, ref SequenceReader<byte> reade
             while (!reader.End)
             {
                 var span = reader.UnreadSpan;
+
+                // Size of header in the current span, if known
+                var length = -1;
+
                 while (span.Length > 0)
                 {
-                    var ch1 = (byte)0;
-                    var ch2 = (byte)0;
-                    var readAhead = 0;
+                    // The size of the EOL terminator. Always 0 (no valid EOL), 1 (LF) or 2 (CRLF)
+                    var eolSize = 0;
```

Comment:
Would it be more obvious to use `-1` as the no-valid-EOL sentinal for consistency?